firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/FirebaseAppDistribution.java

```diff
@@ -147,26 +152,110 @@ public UpdateTask updateIfNewReleaseAvailable() {
               }
               return showUpdateAlertDialog(release);
             })
-        .addOnFailureListener(
-            e -> {
-              postProgressToCachedUpdateIfNewReleaseTask(
-                  UpdateProgress.builder()
-                      .setApkFileTotalBytes(UNKNOWN_RELEASE_FILE_SIZE)
-                      .setApkBytesDownloaded(UNKNOWN_RELEASE_FILE_SIZE)
-                      .setUpdateStatus(UpdateStatus.NEW_RELEASE_CHECK_FAILED)
-                      .build());
-              setCachedUpdateIfNewReleaseCompletionError(
-                  e,
-                  new FirebaseAppDistributionException(
-                      Constants.ErrorMessages.NETWORK_ERROR,
-                      FirebaseAppDistributionException.Status.NETWORK_FAILURE));
-            });
+        .onSuccessTask(
+            unused ->
+                updateApp(true)
+                    .addOnProgressListener(this::postProgressToCachedUpdateIfNewReleaseTask)
+                    .addOnFailureListener(this::setCachedUpdateIfNewReleaseCompletionError))
+        .addOnFailureListener(this::setCachedUpdateIfNewReleaseCompletionError);
 
     synchronized (updateIfNewReleaseTaskLock) {
       return cachedUpdateIfNewReleaseTask;
     }
   }
 
+  private Task<Void> showSignInDialog() {
+    if (isTesterSignedIn()) {
+      return Tasks.forResult(null);
+    }
+
+    TaskCompletionSource<Void> showDialogTask = new TaskCompletionSource<>();
+
+    Activity currentActivity;
+    try {
+      currentActivity = lifecycleNotifier.getNonNullCurrentActivity();
+    } catch (FirebaseAppDistributionException e) {
+      return Tasks.forException(e);
+    }
+
+    signInDialog = new AlertDialog.Builder(currentActivity).create();
+    Context context = firebaseApp.getApplicationContext();
+    signInDialog.setTitle(context.getString(R.string.signin_dialog_title));
+    signInDialog.setMessage(context.getString(R.string.singin_dialog_message));
+
+    signInDialog.setButton(
+        AlertDialog.BUTTON_POSITIVE,
+        context.getString(R.string.singin_yes_button),
+        (dialogInterface, i) -> showDialogTask.setResult(null));
+
+    signInDialog.setButton(
+        AlertDialog.BUTTON_NEGATIVE,
+        context.getString(R.string.singin_no_button),
+        (dialogInterface, i) ->
+            showDialogTask.setException(
+                new FirebaseAppDistributionException(
+                    ErrorMessages.AUTHENTICATION_CANCELED, AUTHENTICATION_CANCELED)));
+
+    signInDialog.setOnCancelListener(
+        dialogInterface ->
+            showDialogTask.setException(
+                new FirebaseAppDistributionException(
+                    ErrorMessages.AUTHENTICATION_CANCELED, AUTHENTICATION_CANCELED)));
+
+    signInDialog.show();
+    return showDialogTask.getTask();
+  }
+
+  private Task<Void> showUpdateAlertDialog(AppDistributionRelease newRelease) {
```

Comment:
Sorry if this is annoying but can you move this back to the same spot in the code where it was before? It would make the diff easier to read.